### ISSUE TITLE
New version monitoring praxis

### ISSUE THREAD

**Issue by iskunk** [chromium-linux-tarballs/chromium-tarballs#9]
Right now, the `cron.yml` workflow is scheduled to run every three hours to check for new releases of Chromium.

This approach has a couple of downsides: (1) up to a 2h59m latency on a new version being processed, and (2) having to start up a relatively heavyweight process, eight times a day, just to run a simple REST API call. (These two traits are, obviously, at odds with each other.)

I'd like to suggest a somewhat different approach: Do the REST call on a cron job on some secure Gentoo server, and if/when a new version is found, use a [repository_dispatch](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#repository_dispatch) call to kick off the tagging workflow right then and there. (It doesn't have to pass the specific version(s) it found; all that matters is the invocation happening at the right time.) This could reasonably be scheduled to run like four or five times an hour.

The GitHub cron scheduling can remain, but that could then be set to run e.g. every eight hours, more as a fallback in case the aforementioned cron job fails than as the primary means of catching new versions.

**Comment by Kangie** [comment]
Sounds reasonable. I'll think about what I have available to host it on - it should be straightforward to just write each channel out to a file and do a string comparison based on that to trigger the `repository_dispatch` call.

**Comment by iskunk** [comment]
Great! Long-term, lightweight monitoring (with a bit of statefulness) is definitely a use case not well served by GitHub CI.

(And thanks for having me onboard :heart:)

**Comment by Kangie** [comment]
Theoretically addressed between 4215cbce87a2d70763550f19eae74c4d1b86ba9e and https://github.com/chromium-linux-tarballs/chromium-ci-trigger

TL;DR:
* CI now checks every 8 hours
* We do a few web requests on a server I control every few hours and (theoretically...) trigger the cron/tagging workflow if there is a change for any channel.

**Comment by iskunk** [comment]
Looks reasonable, just a few remarks:
* I'd suggest specifying something like `RandomizedDelaySec=10m` in the timer unit, so the check doesn't always occur at the same time in the hour;
* By the same token, `cron.yml` would be better off with something like `38 */8 * * *`, as the top of the hour is more than usually busy (since everyone writes their cron schedules with a zero in the first field...)
* Why not use `curl`+`jq` here as well? Keeping the API requests down is especially beneficial in a long-term monitoring job. (For the same load, you could do one call every ten minutes instead of three every thirty minutes)
* Would make sense to wrap the logging routine with a small shell function, given how often it's used.

**Comment by Kangie** [comment]
I'm not _really_ worried about overloading Google's API tbh, but now that we have proof that it works (triggered the overnight stable tarball) we can work on optimisation. Let's track that against the other repo though - this ticket has achieved its purpose.
